DOCUMENTATION.md

```diff
@@ -0,0 +1,115 @@
+```
+
+To preview a single documentation file, copy the content of the
+generated Markdown file into the [preview
+tool](https://registry.terraform.io/tools/doc-preview) on the
+HashiCorp website.
+
+## Templates
+
+Every resource and data source in the provider has its own
+documentation, which includes details of making use of that
+resource/data source, along with the details of its schema.
+
+If you do not write a template for a resource or data source that you
+create, `make generate-docs` will use a basic template when it
+generates documentation for the provider. This template will include
+only the name of the resource/data source and the details of its
+schema. For resources in particular, this is insufficient to guide
+users as it does not include examples of HCL which make use of the
+resource, or instructions for how to import an existing resource into
+the Terraform state.
+
+Note: The documentation generation tool gathers the list of known data
+sources and resources from the provider itself, not by scanning the
+source directory. As a result, if you have created a new data source
+or resource but you have not added into the top-level schema in
+[provider.go](fastly/provider.go) it will not be included in the
+generated documentation. If you create a template for the new data
+source or resources and run `make validate-docs` the validator will
+produce an error as it will be unable to find the new data source or
+resource. The simplest way to avoid this problem is to write and
+execute the acceptance tests for the new data source or resource
+*before* working on its documentation; in order for the acceptance
+tests to pass the new data source or resource must already be included
+in the top-level schema.
+
+Before submitting a PR with your new data source or resource, you
+should add a proper template for it, and also ensure that the schema
+within the data source or resource has been properly documented as
+well. Review the following three sections for details.
+
+## Data Sources
+
+Each data source should have a template file and an example file
+showing basic usage of the data source in HCL. The template for [NGWAF
+Workspaces](templates/data-sources/ngwaf_workspaces.md.tmpl) can be
+used as a reference.
+
+## Resources
+
+Each resource should have a template file and at least two example
+files: one showing basic usage of the data source in HCL, and the
+other demonstrating how to import an existing resource into the
+Terraform state. The examples should be fully functional: they should
+include all resources necessary to create the resource being
+documented. The template for [Config
+Stores](templates/resources/configstore.md.tmpl) can be used as a
+reference.
+
+## Schema Attributes
+
+Every attribute in every schema must include a description. The
+description should provide all of the information necessary for a user
+to understand the purpose and usage of that attribute.
+
+To accomplish this, follow these steps:
```

Comment:
"To write consistent and helpful attribute descriptions, follow these steps:"
(Slightly more precise and user-focused.)